apps/browser/browserservice.cpp

```diff
@@ -23,15 +23,19 @@
 
 #define SAILFISH_BROWSER_UI_SERVICE QLatin1String("org.sailfishos.browser.ui")
 
-#define IS_PRIVILEGED \
+#define GET_PID connection().interface()->servicePid(message().service())
+
+#define IS_PRIVILEGED(reply) \
```

Comment:
and another nit: a macro that looks like a predicate but is a fancy return with double semicolon / other syntactical issues depending on where and how it is used.

well, it was like this originally but imo the end result is that places where it is used are confusing and can't be understood without looking at the macro def

Reply:
Improved these. Now it doesn't take an argument but it's simply a check.